Add Flink 1.20.1 release

afedulov opened the pull request:
Release date will be pinned after the voting for the RC completes.

Commit:
Add Flink 1.20.1 release

afedulov commented:
TODO: add Netty 4 disclaimer.

ferenc-csaky reviewed (approved):
If release date and Netty4 disclaimer will be added, LGTM

davidradl commented on docs/content/posts/2025-02-TBD-release-1.20.1.md:
```diff
@@ -0,0 +1,213 @@
+
+---
+
+The Apache Flink Community is pleased to announce the second bug fix release of the Flink 1.20 series.
```
nit: isn't this the first bug fix release of the Flink 1.20 series?

ferenc-csaky replied:
That is a good catch, it will be. :) I guess 1.19 copy/paste result.

reswqa commented on docs/content/posts/2025-02-TBD-release-1.20.1.md:
IIRC, issues only related to testing shouldn't be listed here.

1996fanrui replied:
+1, FLINK-35512, FLINK-28091, FLINK-37100, FLINK-36591, FLINK-36191 and FLINK-35482 can be removed.

afedulov commented:
Thanks everyone, for the feedback. I addressed your comments, added a site rebuild commit and also included an explicit Netty4 upgrade disclaimer, as was previously discussed: 
<img width="1101" alt="image" src="https://github.com/user-attachments/assets/48b2ebb4-6c7d-4cc2-9f14-ff850ee6a179" />

Commit:
Rebuild site
